src/Git/GitRevLoglist.cpp

```diff
@@ -0,0 +1,239 @@
+// TortoiseGit - a Windows shell extension for easy version control
+
+// Copyright (C) 2008-2015 - TortoiseGit
+
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software Foundation,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+//
+
+#include "stdafx.h"
+#include <ATLComTime.h>
+#include "GitRevLoglist.h"
+#include "Git.h"
+#include "gitdll.h"
+#include "UnicodeUtils.h"
+
+typedef CComCritSecLock<CComCriticalSection> CAutoLocker;
+
+GitRevLoglist::GitRevLoglist(void)
+{
+	GitRev();
+	m_Action = 0;
+	m_RebaseAction = 0;
+	m_IsFull = 0;
+	m_IsUpdateing = 0;
+	m_IsCommitParsed = 0;
+	m_IsDiffFiles = 0;
+	m_CallDiffAsync = nullptr;
+	m_IsSimpleListReady = 0;
+	m_Mark = 0;
+
+	SecureZeroMemory(&m_GitCommit, sizeof(GIT_COMMIT));
+}
+
+GitRevLoglist::~GitRevLoglist(void)
+{
+	if (!m_IsCommitParsed && m_GitCommit.m_pGitCommit)
+		git_free_commit(&m_GitCommit);
+}
+
+void GitRevLoglist::Clear()
+{
+	GitRev::Clear();
+	m_Action = 0;
+	m_Files.Clear();
+	m_Ref.Empty();
+	m_RefAction.Empty();
+	m_Mark = 0;
+
+}
+
+int GitRevLoglist::SafeGetSimpleList(CGit* git)
+{
+	if (InterlockedExchange(&m_IsUpdateing, TRUE) == FALSE)
+	{
+		m_SimpleFileList.clear();
+		git->CheckAndInitDll();
+		GIT_COMMIT commit = { 0 };
+		GIT_COMMIT_LIST list;
+		GIT_HASH   parent;
+
+		CAutoLocker lock(g_Git.m_critGitDllSec);
+
+		try
+		{
+			if (git_get_commit_from_hash(&commit, m_CommitHash.m_hash))

```

Comment:
As we're requesting a new commit object this is also a candidate for using libgit2